micro_sam/training/joint_sam_trainer.py

```diff
@@ -25,15 +25,15 @@ def __init__(
         self.instance_loss = instance_loss
         self.instance_metric = instance_metric
 
-    def save_checkpoint(self, name, best_metric, **extra_save_dict):
+    def save_checkpoint(self, name, current_metric, best_metric, **extra_save_dict):
         current_unetr_state = self.unetr.state_dict()
         decoder_state = []
         for k, v in current_unetr_state.items():
             if not k.startswith("encoder"):
                 decoder_state.append((k, v))
         decoder_state = OrderedDict(decoder_state)
 
-        super().save_checkpoint(name, best_metric, decoder_state=decoder_state, **extra_save_dict)
+        super().save_checkpoint(name, current_metric, best_metric, decoder_state=decoder_state, **extra_save_dict)
```

Comment:
Can you pass `current_metric` and `best_metric` by keyword here? I think the current code (passing them as positional arguments) will lead to a wrong result until we have the new version of `torch_em` here.

Reply:
And passing the arguments by keyword will work with the new and old version.

Reply:
Fixed. Thanks for spotting!